docs/newFeature.md

```diff
@@ -41,9 +45,127 @@ The first function, `OnNewMessage`, implements the feature's reactions to the us
+    myFeatures = []
+
+class MyFeature:
+    def __init__(self, agentId, conversationId):
+        self.agentId = agentId
+        self.conversationId = conversationId
+
+        # Create and connect the Conversation handler and add the OnNewMessage callback.
+        self.handlerConversation = ConversationNotificationBusHandler()
+        self.handlerConversation.connect(self.connectionId)
+        self.handlerConversation.add_callback("OnNewMessage", self.OnNewMessage)
+
+        # Create and connect the Agent handler and add the OnAIResponse callback.
+        self.handlerAgent = AIAgentNotificationBusHandler()
+        self.handlerAgent.connect(self.connectionId)
+        self.handlerAgent.add_callback("OnAIResponse", self.OnAIResponse)
+
+    # This function is called when a new message is received from the user (chat widget).
+    def OnNewMessage(self, message):
+        # The message is a list of length, where the first element is the message from the user.
+        prompt = [{"role": "user", "content": [message[0]]}]
+
+        # A call to the agent can be made to get the history of the conversation.
+        # The history is a json formatted string consisting of a list of messages with roles and content.
+        history = AIAgentRequestBus(bus.Event, "GetHistory", self.agentId)
+        jsonHistory = json.loads(history)
+
+        # The prompt can be modified and sent to the agent. The format of the prompt is a list of messages with roles and content.
+        fullJsonStringPrompt = "..."
+        AIAgentRequestBus(bus.Event, "SendPrompt", self.agentId, fullJsonStringPrompt)
+
+    # This function is called when the response from the AI agent is received.
+    def OnAIResponse(self, message):
+        # The message is a list of length 1, where the first element is the response from the AI agent.
+        jsonMessage = json.loads(message[0])
+
+        response = "..."
+        summary = ["...", "...", ...]
+
+        # The response can be further processed, sent back to the agent or the process can finish by calling the OnFeatureResponse.
+        # This function notifies the UI that the response is ready to be displayed and a new message can be sent. This always needs to be called.
+        ConversationNotificationBus(bus.Event, "OnFeatureResponse", self.connectionId, response, summary)
+
+# It is important to add each feature as a separate instance to avoid redefinitions of the same feature and mismatching of the ids used for busses.
+myFeatures.append(MyFeature(agentId, conversationId))
+```
+It is important to create the feature as a class and store its objects in some kind of data structure (like the `myFeatures` list). This script is called each
+time a new conversation is created with this feature and creating a class object prohibits overriding the previously declared function. This is done because of a limitation of Python interpreter in O3DE. The interpreter is global and all the scripts are executed in the same context.  
```

Comment:
```suggestion
It is important to create the feature as a class and store its objects in some kind of data structure (like the `myFeatures` list). This script is called each time a new conversation is created with this feature and creating a class object prohibits overriding the previously declared function. This is done because of a limitation of the Python interpreter in O3DE. The interpreter is global and all scripts are executed in the same context.  
```